sws.go

```diff
@@ -56,6 +56,7 @@ func main() {
 		log.Infof("No configuration file specified. Will rely on environment for configuration.")
 		Configuration = config.NewConfig()
 	}
+	config.SWSConfig = Configuration
```

Comment:
What is this for? I assume you are trying to make a global variable? I was trying to keep from creating global variables -  sometimes this ends up causing circular dependencies in imports (at least in my experience with Go programming).

Did you try to pass the Configuration object around? That is what I've been trying to do. Seems odd in some places we are passing it around and in others we are using this new global config.SWSConfig. If you feel we need a global, then refactor sws.go and use config.SWSConfig rather than Configuration variable found within sws.go (though, again, I think we should avoid expecting to use global variables through code - that will get messing as the code base grows).

Reply:
> What is this for? I assume you are trying to make a global variable?

Yes, I guess a global variable makes sense. As we should pass the config through several layers:
main -> server -> handler -> backend -> helpers -> ...

So passing a config in all signature methods sounds to me something we can avoid.
I agree about circular dependencies, but if config package remains simple, then the global variable can keep simple and useful (I'm thinking more like the version.* pattern, when version is a global variable defined in a well located package).

I'm +1 with your comment, I can refactor the sws.go to use global variable at all.
But I didn't want to add it before some chat about it. I guess on this case circular dependencies are a low risk.

Reply:
Sorry to comment after your refactor. But I just wanted to remark that although it makes things faster to write global states can in the end increase complexity, increase side effects, make things like unit tests harder to write. So I hope we will keep number of globals low :)

Reply:
Yes, I guess we are in the same page.
BUT in an app where we have a single server, I don't see correct to propagate the config in the signature as we will have more than 4 or 5 layers.
Sorry, but that's unnecessary.
Config will be loaded when program starts and will be used for specific use cases.
So, that's reason I guess it makes more sense.

Like the version of a program, there is a global variable for it, I guess we are talking about that, not create singleton patterns because are cheap.